### PR TITLE
Move VW object creation to sklearn constructor

### PR DESCRIPTION
**Pull request by Sharad24**
Reference Issue #1469

### CODE REVIEW

File: python/vowpalwabbit/sklearn_vw.py [hunk 1]
```diff
@@ -508,14 +508,31 @@ class VWClassifier(SparseCoefMixin, ThresholdingLinearClassifierMixin, VW):
     note - don't try to apply link='logistic' on top of the existing functionality
     """
 
-    def __init__(self, **params):
+    def __init__(self, X=None, y=None, sample_weight=None, **params):
```
Comment (jackgerrits) [inline python/vowpalwabbit/sklearn_vw.py]: Why is init taking the features and labels? What if you want to predict/learn with different examples?

Comment (Sharad24) [reply python/vowpalwabbit/sklearn_vw.py]: My initial thought was if the data is passed the initialized model could be trained as well. If its not passed, user can still call fit/predict. Do we only require object creation or should it be training+testing in the constructor requiring X_train, y_train, X_test, y_test with defaults as None?

Comment (jackgerrits) [reply python/vowpalwabbit/sklearn_vw.py]: The constructor should not be training the model

**Comment by jackgerrits** [comment]
@Scott-Graham-Bose does this seem good to you?

**Comment by jackgerrits** [comment]
I'm not sure if you noticed @Sharad24, @gramhagen commented on the issue. It looks like we do actually want to keep the lazy initialization. Here is @gramhagen's response quoted below:

> @jackgerrits I dug in a bit more on item 1. I believe lazy initialization is the right construct for this estimator, and it was added to support sklearn pipelines. We should keep the vw model construction out of init to maintain support for this. Handling the case where a data file is provided makes this a little funky, but I think it's better to stick to a single conceptual model, where training only happens when fit is called (and the vw model only needs to be instantiated at this point).

**Comment by Sharad24** [comment]
Yes, I saw. It definitely makes sense to allow for building pipelines in sklearn. I'll revoke those changes. I'll push changes with point 4 in the issue, on addition of the `convert_labels` parameter.
